code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -221,17 +221,25 @@
 //Medical Stuff
 /obj/item/borg/upgrade/medical
 	name = "medical cyborg MK-2 upgrade board"
-	desc = "Used to give a medical cyborg advanced care tools."
+	desc = "Used to give a medical cyborg advanced care tools. Also increases storage capacity for medical consumables."
 	icon_state = "cyborg_upgrade"
 	required_modules = list(MEDICAL_MODULE, SYNDIE_CRISIS_MODULE)
 	modules_to_add = list(/obj/item/weapon/melee/defibrillator,/obj/item/weapon/reagent_containers/borghypo/upgraded)
+	var/increases_storage = 1
+
+/obj/item/borg/upgrade/medical/attempt_action(var/mob/living/silicon/robot/R, var/mob/living/user)
+	if(..())
+		return FAILED_TO_ADD
+	if(increases_storage)
+		R.module.respawnables_max_amount = MEDICAL_MAX_KIT * 2
 
 /obj/item/borg/upgrade/medical/organ_gripper
 	name = "medical cyborg organ gripper upgrade"
 	desc = "Used to give a medical cyborg a organ gripper."
 	icon = 'icons/obj/device.dmi'
 	icon_state = "gripper-medical"
 	modules_to_add = list(/obj/item/weapon/gripper/organ)
+	increases_storage = 0
```

Comment:
instead of doing this, is there any good reason to not just repath this to be its own thing instead of a child of the MK2?

you would have to add a required_modules list for it but that's overall 1 less line and it avoids weird vars like this

Reply:
I was simply too lazy at the time to repath everything, will fix